### PR TITLE
Yab buildfactory

### PR DESCRIPTION
**Pull request by lorglas**

### CODE REVIEW

**Comment by Begasus** [comment]
Can't seem to find the trailing whitespace here, anyone got an idea?

File: haiku-apps/yab_buildfactory/yab_buildfactory-2.4.recipe [hunk 1]
```diff
@@ -0,0 +1,44 @@
+SUMMARY="An buildfactory for yab programs"
```
Comment (humdingerb) [inline haiku-apps/yab_buildfactory/yab_buildfactory-2.4.recipe:1]:
```suggestion
SUMMARY="A buildfactory for yab programs"
```

File: haiku-apps/yab_buildfactory/yab_buildfactory-2.4.recipe [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+DESCRIPTION="The buildfactory allows you to generate standalone programs."
+HOMEPAGE="http://software.besly.de/"
```
Comment (humdingerb) [inline haiku-apps/yab_buildfactory/yab_buildfactory-2.4.recipe:3]:
```suggestion
HOMEPAGE="https://software.besly.de/"
```

File: haiku-apps/yab_buildfactory/yab_buildfactory-2.4.recipe [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	2015-2017 Jim Saxton
+	2020 Lorenz (lorglas) BeSly Software Solutions"
+LICENSE="Artistic"
+REVISION="1"
+SOURCE_URI="https://github.com/lorglas/yab_buildfactory/archive/$portVersion.tar.gz"
+CHECKSUM_SHA256="c1bf29537fcb1a6988ca81bcad1485532090ddd5f11fe962a68c40f927822489"
+DISABLE_SOURCE_PACKAGE="YES"
+ARCHITECTURES="x86_gcc2 !x86 x86_64"
+ADDITIONAL_FILES="yab_buildfactory.sh
+yab.hvif"
```
Comment (humdingerb) [inline haiku-apps/yab_buildfactory/yab_buildfactory-2.4.recipe:12]: Add another tab in front.

**Comment by lorglas** [comment]
i make the changes

File: haiku-apps/yab_buildfactory/yab_buildfactory-2.4.recipe [hunk 4]
```diff
@@ -0,0 +1,44 @@
+SUMMARY="A buildfactory for yab programs"
+DESCRIPTION="The buildfactory allows you to generate standalone programs."
+HOMEPAGE="https://software.besly.de/"
+COPYRIGHT="2006-2015 Jan Bungeroth
+	2015-2017 Jim Saxton
+	2020 Lorenz (lorglas) BeSly Software Solutions"
+LICENSE="Artistic"
+REVISION="1"
+SOURCE_URI="https://github.com/lorglas/yab_buildfactory/archive/$portVersion.tar.gz"
+CHECKSUM_SHA256="c1bf29537fcb1a6988ca81bcad1485532090ddd5f11fe962a68c40f927822489"
```
Comment (Begasus) [inline haiku-apps/yab_buildfactory/yab_buildfactory-2.4.recipe:10]: Maybe add SOURCE_FILENAME here?:
```diff
+SOURCE_FILENAME="yab_buildfactory-$portVersion.tar.gz"
```

File: haiku-apps/yab_buildfactory/yab_buildfactory-2.4.recipe [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+ADDITIONAL_FILES="yab_buildfactory.sh
+	yab.hvif"
+ARCHITECTURES="x86_gcc2 !x86 x86_64"
+DISABLE_SOURCE_PACKAGE="YES"
+POST_INSTALL_SCRIPTS="$relativePostInstallDir/yab_buildfactory.sh"
```
Comment (Begasus) [inline haiku-apps/yab_buildfactory/yab_buildfactory-2.4.recipe:15]: Some reordering should be done here, check https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#ordering

File: haiku-apps/yab_buildfactory/yab_buildfactory-2.4.recipe [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+PROVIDES="
+	yab_buildfactory = $portVersion
+	app:yab_buildfactory = $portVersion
+	"
+REQUIRES="
+	haiku
+	yab
+	yab_source
```
Comment (Begasus) [inline haiku-apps/yab_buildfactory/yab_buildfactory-2.4.recipe:23]: Is this provided by any package? Can't seem to find it (yab_source)

File: haiku-apps/yab_buildfactory/yab_buildfactory-2.4.recipe [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+	"
+#devel:libz is needed both to build and for the BuildFactory to do its work.
+
+BUILD_REQUIRES=""
+BUILD_PREREQUIRES=""
+
+INSTALL()
+{
+	mkdir -p $appsDir/yab-buildfactory
+	addattr -t icon -f $portDir/additional-files/yab.hvif "BEOS:ICON" "BeSly_Buildfactory.yab"
```
Comment (bbjimmy) [inline haiku-apps/yab_buildfactory/yab_buildfactory-2.4.recipe:37]: There is no aditional-files directory in yab_buildfactory-$portVersion.tar.gz-$portVersion.tar.gz

Comment (Begasus) [reply haiku-apps/yab_buildfactory/yab_buildfactory-2.4.recipe:37]: Additional-files are supplied in this PR (hence $portDir/additional-files)

**Pull request merged**